### PR TITLE
Added Initial Metadata Support.

### PR DESCRIPTION
**Pull request by iamkirkbater**
Hi.  So one thing I saw that I personally was missing from version 2.x of Winston was the metadata object, so I created a custom formatter to accommodate this.  I have tests written in mocha/should (as it was originally created for work, so I had to make it work with our systems, but I'm in the process of backporting it, and offer it to the official Winston project.  If you don't want it in the winston project itself, I can always just create a spinoff library for this as well.  

Before I spend all the time writing the tests, though, I figured I'd introduce the idea and see if you liked it or not.  I assume you won't accept the PR without tests, and I have no problem with that, I just don't want to spend all the time writing the tests if you're not interested in the formatter :).

You'll note that I did add an example.js which you can run to see the output, but essentially it ends up logging out this:

```
$ node examples/metadata.js
{ level: 'info',
  message: 'This should be a message.',
  metadata:
   { application: 'Microsoft Office',
     store: 'Big Box Store',
     purchaseAmount: '9.99',
     timestamp: '2018-02-19T21:57:09.989Z' } }
{ message: 'This should have attached metadata',
  level: 'info',
  timestamp: '2018-02-19T21:57:10.001Z',
  metadata: { category: 'movies', subCategory: 'action' } }
{ level: 'debug',
  message: 'This message should be outside of the bookInfo object',
  label: 'myCustomLabel',
  timestamp: '2018-02-19T21:57:10.002Z',
  bookInfo:
   { publisher: 'Lorem Press',
     author: 'Albert Einstein',
     book: '4D Chess for Dummies' } }
{ level: 'debug',
  message: 'This message should be outside of the bookInfo object',
  label: 'myCustomLabel',
  timestamp: '2018-02-19T21:57:10.002Z',
  metadata:
   { bookInfo:
      { publisher: 'Lorem Press',
        author: 'Albert Einstein',
        book: '4D Chess for Dummies' },
     transactionInfo:
      { purchasePrice: '9.99',
        purchaseDate: '2.10.2018',
        transactionId: '123ABC' } } }
```

Let me know if there's anything you'd like to see modified, or if you like it and just want the tests written and I can work on porting them over into your test suite.

Thanks!
Kirk

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "logform",
-  "version": "1.2.2",
+  "version": "1.3.0",
```
Comment (indexzero) [inline package.json]: Please do not include version bumps in PRs. As a maintainer team we may merge multiple features into a future minor version bump.

**Comment by indexzero** [comment]
@iamkirkbater definitely seems like a feature the project would want. There are enough folks who prefer to nest part (or all) of their log information inside a `meta` or `metadata` object that it should be officially possible, but not the default.

File: package.json (hunk 1 above)
Comment (iamkirkbater) [reply package.json]: No problem.  This makes total sense, I'm just used to doing it haha.  I'll edit accordingly.

**Comment by iamkirkbater** [comment]
Excellent, great to hear.  I'll work on porting over the tests and I'll revert the change to the version file and hopefully I can push those up by Wednesday or Thursday if I don't get distracted at work and just do it then :)

**Comment by iamkirkbater** [comment]
@indexzero I actually finished up the work I was going to do on something else a lot earlier tonight than I had expected, so I just got all the changes rolled in.  Let me know if there's anything else you need from me on this :)

**Comment by indexzero** [comment]
Awesome thanks @iamkirkbater – going to merge this and release it today.

**Pull request merged**